apps/antalmanac/src/providers/Theme.tsx

```diff
@@ -12,25 +11,20 @@ interface Props {
  * sets and provides the MUI theme for the app
  */
 export default function AppThemeProvider(props: Props) {
-    const [darkMode, setDarkMode] = useState(isDarkMode());
+    const { theme, setTheme } = useThemeStore();
 
     useEffect(() => {
-        AppStore.on('themeToggle', () => {
-            setDarkMode(isDarkMode());
-        });
         window.matchMedia('(prefers-color-scheme: dark)').addEventListener('change', (e) => {
-            if (AppStore.getTheme() === 'auto') {
-                setDarkMode(e.matches);
-            }
+            setTheme(e.matches ? 'dark' : 'light');
```

Comment:
Could you fix this? You have to remove the event listener once the component unmounts. I don't have some of the TypeScript definitions on hand, so you'll have to find them.

```tsx
const mediaQueryList =  window.matchMedia('(prefers-color-scheme: dark)')

const onChange = (e) => {
  setTheme(e.matches ? 'dark' : 'light');
}

mediaQueryList.addEventListener('change', onChange)

return () => {
  mediaQueryList.removeEventListener('change', onChange)
}
```